backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/descriptor/factory/standard/GridTabVOBasedDocumentEntityDescriptorFactory.java

```diff
@@ -105,6 +108,9 @@
 	// Services
 	private static final Logger logger = LogManager.getLogger(GridTabVOBasedDocumentEntityDescriptorFactory.class);
 	private final transient IColumnBL adColumnBL = Services.get(IColumnBL.class);
+
+	private final Collection<IDocumentDecorator> documentDecorators = SpringContextHolder.instance.getBeansOfType(IDocumentDecorator.class);
```

Comment:
consider passing it as ctor argument, same as `@NonNull final LookupDataSourceFactory lookupDataSourceFactory`